src/App.js

```diff
@@ -1,19 +1,143 @@
 import React from 'react';
 import './App.css';
+import { TodoList } from './components/TodoList';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
-  );
+const preparedTodos = todos.map(todo => ({
+  ...todo,
+  user: users.find(user => user.id === todo.userId),
+}));
+
+class App extends React.Component {
+  state = {
+    todos: preparedTodos,
+    user: '',
+    title: '',
+    isCorrectTitle: false,
+    isCorrectUserName: false,
+  }
+
+  addTodo = () => {
+    this.setState(state => ({
+      todos: [...state.todos, {
+        userId: users.find(user => user.name === state.user).id,
+        id: state.todos[state.todos.length - 1].id + 1,
+        title: state.title,
+        completed: false,
+        user: users.find(user => user.name === state.user),
+      }],
+    }));
+  }
+
+  hendlerFormChange = (event) => {
+    const { user, title } = this.state;
+
+    event.preventDefault();
+    this.setState(state => ({
+      isCorrectUserName: !user,
+    }));
+
+    if (title.length < 3) {
+      this.setState({
+        isCorrectTitle: true,
+      });
+
+      return;
+    }
+
+    if (!user) {
+      return;
+    }
+
+    this.addTodo();
+    this.setState({
+      user: '',
+      title: '',
+    });
+  }
+
+  render() {
+    const {
+      isCorrectTitle,
+      isCorrectUserName,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>Add todo form</h1>
+
+        <p>
+          <span>Users: </span>
+          {users.length}
+        </p>
+        <form
+          className="form"
+          onSubmit={this.hendlerFormChange}
+        >
+          <label
+            htmlFor="title"
+          >
+            Title
+          </label>
+          <input
+            value={this.state.title}
+            id="title"
+            onChange={(event) => {
+              this.setState({
+                title: event.target.value,
+                isCorrectTitle: false,
+              });
+            }}
+          />
+          {isCorrectTitle && (
+            <span className="error">
+              please enter a title longer than 3 characters
+            </span>
+          )}
```

Comment:
The name `isCorrectTitle` means that title is correct. But in fact, you show an error if it's true.
You should show an error when `!isCorrectTitle`